scipy/_lib/_array_api.py

```diff
@@ -652,3 +657,139 @@ def xp_default_dtype(xp):
 def is_marray(xp):
     """Returns True if `xp` is an MArray namespace; False otherwise."""
     return "marray" in xp.__name__
+
+
+
+def _make_capabilities(skip_backends=None, cpu_only=False, np_only=False,
+                       exceptions=None):
+    skip_backends = [] if skip_backends is None else skip_backends
+    exceptions = [] if exceptions is None else exceptions
+
+    capabilities = dataclasses.make_dataclass('capabilities', ['cpu', 'gpu', 'mask'])
+
+    # Default capabilities
+    numpy = capabilities(cpu=True, gpu=False, mask=True)
+    strict = capabilities(cpu=True, gpu=False, mask=False)
+    cupy = capabilities(cpu=False, gpu=True, mask=False)
+    torch = capabilities(cpu=True, gpu=True, mask=False)
+    jax = capabilities(cpu=True, gpu=True, mask=False)
+    dask = capabilities(cpu=True, gpu=True, mask=False)
+
+    capabilities = dict(numpy=numpy, array_api_strict=strict, cupy=cupy,
+                        torch=torch, jax=jax, dask=dask)
+
+    for backend, _ in skip_backends:  # ignoring the reason
+        setattr(capabilities[backend], 'cpu', False)
+        setattr(capabilities[backend], 'gpu', False)
+        setattr(capabilities[backend], 'mask', False)
+
+    other_backends = {'cupy', 'torch', 'jax', 'dask'}
+
+    if cpu_only:
+        for backend in other_backends - set(exceptions):
+            setattr(capabilities[backend], 'gpu', False)
+
+    if np_only:
+        numpy.mask = False
+        for backend in other_backends:
+            setattr(capabilities[backend], 'cpu', False)
+            setattr(capabilities[backend], 'gpu', False)
+
+    return capabilities
+
+
+def _make_capabilities_table(capabilities):
+    numpy = capabilities['numpy']
+    cupy = capabilities['cupy']
+    torch = capabilities['torch']
+    jax = capabilities['jax']
+    dask = capabilities['dask']
+
+    for backend in [numpy, cupy, torch, jax, dask]:
+        for attr in ['cpu', 'gpu', 'mask']:
+            val = "✓" if getattr(backend, attr) else " "
+            val += " " * (len('{numpy.}') + len(attr) - 1)
+            setattr(backend, attr, val)
+
+    table = f"""                
+    +---------+-------------+-------------+--------------+
+    | Library | CPU         | GPU         | ``marray``   |
+    +=========+=============+=============+==============+
+    | NumPy   | {numpy.cpu} | ✗           | {numpy.mask} |
+    +---------+-------------+-------------+--------------+
+    | CuPy    | ✗           | {cupy.gpu } | {cupy.mask } |
+    +---------+-------------+-------------+--------------+
+    | PyTorch | {torch.cpu} | {torch.gpu} | {torch.mask} |
+    +---------+-------------+-------------+--------------+
+    | JAX     | {jax.cpu  } | {jax.gpu  } | {jax.mask  } |
```

Comment:
I think it's important to distinguish between JAX in eager mode vs. jax.jit

Reply:
Would there be a column or row for that? I was aware of the distinction, but I don't know much more.

Reply:
jax.jit functionality is always a subset of eager jax. So maybe the layout could look something like this?
```
JAX | {green check mark} |
JAX | {yellow warning symbol} Eager only |
JAX | {red x} |
```

Reply:
Sure. I was hoping to start simple, though. Would this require a change to the information stored in the `xp_capabilities_table ` dictionary, or do you mean below that it be looked up for a given function with `lazy_xp_function`?